app/services/github_service.rb

```diff
@@ -74,17 +75,63 @@ def get_all_posts(oauth_token)
       posts = client.contents(full_repo_name, path: '_posts')
       posts.each do |post|
         oldest_commit = get_oldest_commit_for_file(client, post.path)
-
-        if client.user[:login] == oldest_commit[:author][:login]
+        username = client.user[:login]
+        if username == oldest_commit[:author][:login]
           post_api_response = client.contents(full_repo_name, path: post.path)
-          # Base64.decode64 will convert our string into a ASCII string
-          # calling force_encoding('UTF-8') will fix that problem
-          text_contents = Base64.decode64(post_api_response.content).force_encoding('UTF-8')
-          result << PostFactory.create_post(text_contents, post.path)
+
+          post_model = create_post_from_api_response(post_api_response, nil)
+          image_paths = KramdownService.get_all_image_paths(post_model.contents)
+
+          images = []
+          image_paths.each do | image_path |
+            image_content = client.contents(full_repo_name, path: image_path)
+            images << create_post_image(image_path, image_content.content)
+          end
+
+          post_model.images = images
+          
+          result << post_model
         end
       end
       result
     end
+
+    ##
+    # This method fetches all of the posts that a user has written but have not been merged into master yet.
+    #
+    # Params:
+    # +oauth_token+::a user's oauth access token
+    def get_all_posts_in_pr_for_user(oauth_token)
+      result = []
+      client = Octokit::Client.new(access_token: oauth_token)
+      pull_requests_for_user = get_open_post_editor_pull_requests(oauth_token)
+      
+      pull_requests_for_user.each do | pull_request |
+        pull_request_files = client.pull_request_files(full_repo_name, pull_request[:number])
+
+        post = nil
+        images = []
+        pull_request_files.each do | pull_request_file |
+          contents_url_params = CGI.parse(pull_request_file[:contents_url])
+
+          # The CGI.parse method returns a hash with the key being the URL and the value being an array of
```

Comment:
Did you mean "returns a map" instead of "returns a hash"?

Reply:
No I mean Hash. Remember a hash in ruby is the map data type.